FIX: Don't overwrite computed property for loading spinner fix

eviltrout opened the pull request:
This fixes an issue CvX found on PR #14666 where a previous fix
overwrote a computed property.

The better fix (as is often the case with Ember) is to remove an
observer and call methods when things change ourselves.

<!-- NOTE: All pull requests should have tests (rspec in Ruby, qunit in JavaScript). If your code does not include test coverage, please include an explanation of why it was omitted. -->

eviltrout commented on app/assets/javascripts/discourse/app/routes/discovery.js:
```diff
@@ -46,25 +46,21 @@ export default DiscourseRoute.extend(OpenComposer, {
       if (!this.session.get("topicListScrollPosition")) {
         scrollTop();
       }
-      return false;
```
These `return true` things are not necessary in actions. I'm not sure they ever were. Probably wishful thinking!

CvX commented:
Two notes about the commit message 😅
1. "comptued" typo
2. and could you remove the "@" before my nick? 😄 I learned some time ago that there are a lot of forks of this repo, and I'm gonna get a notification for each one of them (when they pull `main`)

eviltrout commented:
done

CvX commented:
Hm, I see some failures? Unrelated?

And re: commit message, I think when you squash-merge it will suggest the text from the commit, not PR body. 😅

eviltrout commented:
The failure seemed legit! Apparently we were depending on loading not bubbling from the action to the next loading event. I've fixed that and corrected the commit message.
